### PR TITLE
Fixing bugs 

### PR DESCRIPTION
**Pull request by Misha7776**
**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added
- [ ] screenshot of feature is added if task has UI

### CODE REVIEW

File: app/controllers/account/users_controller.rb [hunk 1]
```diff
@@ -1,14 +1,15 @@
 class Account::UsersController < ApplicationController
   before_action :authenticate_user!
-  before_action :authorize_admin!
+  before_action :authorize_admins!
 
   def index
-    @search = current_tenant.users.ransack(params[:q])
+    users = current_user.super_admin? ? super_admin_collection : admin_collection
```
Comment (romaslmd) [inline app/controllers/account/users_controller.rb]: move all these to a so-called Service

File: app/views/account/shared/_nav.html.slim [hunk 2]
```diff
@@ -7,26 +7,19 @@
             i.ft-home
             span
               | Dashboard
-        - if current_user.admin?
+        - if current_user.admin? || current_user.super_admin?
```
Comment (romaslmd) [inline app/views/account/shared/_nav.html.slim]: you are using this condition `admin? && super_admin?` in so many cases.
mb move it to a separate method on model? (e.g.: `current_user.god?`)

File: app/controllers/account/users_controller.rb (hunk 1 above)
Comment (igorkasyanchuk) [inline app/controllers/account/users_controller.rb]: create method collection
and return there:
current_user.super_admin? ? super_admin_collection : admin_collection

File: app/controllers/account/users_controller.rb [hunk 3]
```diff
@@ -64,11 +65,23 @@ def resource_params
...
+
+  def super_admin_collection
+    User.all.by_date
+  end
+
+  def super_admin_resource
+    super_admin_collection.find(params[:id])
+  end
+
+  def determine_resource
```
Comment (igorkasyanchuk) [inline app/controllers/account/users_controller.rb]: def resource
  collection.find(params[:id])

and do change with collection requested above

File: app/views/account/users/_search_form.html.slim [hunk 4]
```diff
@@ -1,19 +1,14 @@
...
-              include_blank: true,
-              class: 'col-md-7 form-control'
-
-    .form-group.row
-      = f.submit 'Filter', class: 'offset-md-4 col-md-1 btn btn-success'
-      = link_to 'Reset', account_users_path, class: 'offset-md-5 col-md-1 btn btn-danger'
\ No newline at end of file
+= search_form_for search, url: account_users_path do |f|
+  .form-group.row
+    = f.search_field :first_name_or_last_name_or_email_cont,
```
Comment (igorkasyanchuk) [inline app/views/account/users/_search_form.html.slim]: make filter in one line https://i.stack.imgur.com/TYl4G.png

Comment (romaslmd) [reply app/views/account/users/_search_form.html.slim]: kinda nice field name :)

Comment (igorkasyanchuk) [reply app/views/account/users/_search_form.html.slim]: This is from ransack gem) it's ok

File: spec/controllers/account/users_controller_spec.rb [hunk 5]
```diff
@@ -189,4 +189,30 @@
       expect(response).to redirect_to root_path
     end
   end
+
+  describe 'Super_admin collection and resourse' do
```
Comment (DimaSmagulov) [inline spec/controllers/account/users_controller_spec.rb]: What is this name means? You are testing 2 actions (index and show) in one test. Move this tests upper into show and index tests with proper context names please.

**Pull request merged**